### PR TITLE
Update map force position

### LINKED ISSUE #53: Question: Zoom level for Electoral map
Hello,
The default [zoom level](https://github.com/localgovdrupal/localgov_elections_reporting/blob/f10be82663d5a48ff967a8e697120f9676605c0d/config/install/views.view.electoral_map.yml#L415) is appearing too small in the Electoral map.  The [map centre](https://github.com/localgovdrupal/localgov_elections_reporting/blob/f10be82663d5a48ff967a8e697120f9676605c0d/config/install/views.view.electoral_map.yml#L411) is also set in Cumbria.  I had to manually adjust this in the [electoral_map view](https://github.com/localgovdrupal/localgov_elections_reporting/blob/1.x/config/install/views.view.electoral_map.yml) for my location.  Is that the right way to do this or have I missed something?

FOA @chriswales95  @dedavidson @finnlewis

### PR DESCRIPTION
**Pull request by dedavidson**
Disable positing forcing in the map view so leafet's auto-boxing runs. Also add update for anyone who has already installed.

### CODE REVIEW

**Comment by finnlewis** [comment]
@Adnan-cds  is testing this.

File: config/install/views.view.electoral_map.yml [hunk 1]
```diff
@@ -407,7 +407,7 @@ display:
             options: '{"position":"topright","setView":"untilPanOrZoom","returnToPrevBounds":true,"keepCurrentZoomLevel":true,"strings":{"title":"Locate my position"}}'
             automatic: false
           map_position:
-            force: true
+            force: false
```
Comment (Adnan-cds) [inline config/install/views.view.electoral_map.yml:410]: This works great.  Thanks a lot :)

File: localgov_elections_reporting.install [hunk 2]
```diff
@@ -181,3 +181,29 @@ function localgov_elections_reporting_update_9003(&$sandbox) {
...
+  foreach ($config_to_update as $config) {
+    $data = $install_dir->read($config);
+    $config_obj = \Drupal::configFactory()->getEditable($config);
+    $config_obj->setData($data);
+    $config_obj->save(TRUE);
+  }
+
+  Drupal::logger('localgov_elections_reporting')->notice("Updated map to use auto-boxing");
+
+  drupal_flush_all_caches();
```
Comment (Adnan-cds) [inline localgov_elections_reporting.install]: I don't think we need to be this explicit.  `drush updatedb` clears the Drupal cache by default anyway.

File: localgov_elections_reporting.install [hunk 3]
```diff
@@ -181,3 +181,29 @@ function localgov_elections_reporting_update_9003(&$sandbox) {
...
+
+  $config_to_update = [
+    'views.view.electoral_map',
+  ];
+
+  $install_dir = new FileStorage($module_path . '/config/install/');
+  foreach ($config_to_update as $config) {
+    $data = $install_dir->read($config);
+    $config_obj = \Drupal::configFactory()->getEditable($config);
+    $config_obj->setData($data);
```
Comment (Adnan-cds) [inline localgov_elections_reporting.install]: This overwrites existing config which leads to loss of UUID, etc.  I would suggest something like this:
```
$data = $config_obj->getRawData()
$data['display']['default']['display_options']['style']['options']['map_position']['force'] = FALSE;
$config_obj->setData($data)
$config_obj->save(TRUE);
```

**Review by Adnan-cds — CHANGES_REQUESTED** [review]
Thanks for looking into this so promptly.  I have spotted a minor issue with the update hook implementation which I have pointed out.

File: localgov_elections_reporting.install (hunk 2 above)
Comment (dedavidson) [reply localgov_elections_reporting.install]: Have removed

File: localgov_elections_reporting.install (hunk 3 above)
Comment (dedavidson) [reply localgov_elections_reporting.install]: Reworked this.

**Review by Adnan-cds — APPROVED** [review]
Works perfect.  Thanks a lot :)

**Pull request merged**